examples/01-filter/image-fft-perlin-noise.py

```diff
@@ -0,0 +1,206 @@
+# no physical meaning in the physical domain. PyVista will drop the imaginary
+# content, but will warn you of it.
+#
+# As expected, we only see low frequency noise.
+
+low_pass = sampled_fft.low_pass(1.0, 1.0, 1.0).rfft()
+low_pass['scalars'] = np.real(low_pass.active_scalars)
+warped_low_pass = low_pass.warp_by_scalar()
+warped_low_pass.plot(show_scalar_bar=False, text='Low Pass of the Perlin Noise', lighting=False)
+
+
+###############################################################################
+# High Pass Filter
+# ~~~~~~~~~~~~~~~~
+# This time, let's perform a high pass filter on the frequency content and then
+# convert it back into the space (pixel) domain by immediately applying a
+# reverse FFT.
+#
+# When converting back, keep only the real content. The imaginary content has no
+# physical meaning in the pixel domain.
+#
+# As expected, we only see the high frequency noise content as the low
+# frequency noise has been attenuated.
+
+high_pass = sampled_fft.high_pass(1.0, 1.0, 1.0).rfft()
+high_pass['scalars'] = np.real(high_pass.active_scalars)
+warped_high_pass = high_pass.warp_by_scalar()
+warped_high_pass.plot(show_scalar_bar=False, text='High Pass of the Perlin Noise', lighting=False)
+
+
+###############################################################################
+# Sum Low and High Pass
+# ~~~~~~~~~~~~~~~~~~~~~
+# Show that the sum of the low and high passes equal the original noise.
+
+grid = pv.UniformGrid(dims=sampled.dimensions, spacing=sampled.spacing)
+grid['scalars'] = high_pass['scalars'] + low_pass['scalars']
+
+print(
+    'Low and High Pass identical to the original:', np.allclose(grid['scalars'], sampled['scalars'])
+)
+
+pl = pv.Plotter(shape=(1, 2))
+pl.add_mesh(sampled.warp_by_scalar(), show_scalar_bar=False, lighting=False)
+pl.add_text('Original Dataset')
+pl.subplot(0, 1)
+pl.add_mesh(grid.warp_by_scalar(), show_scalar_bar=False, lighting=False)
+pl.add_text('Sum of the Low and High Passes')
+pl.show()
+
+
+###############################################################################
+# Animate
+# ~~~~~~~
+# Animate the variation of the cutoff frequency.
+
+
+def warp_low_pass_noise(cfreq, scalar_ptp=abs(sampled['scalars']).ptp()):
+    """Process the sampled FFT and warp by scalars."""
+    output = sampled_fft.low_pass(cfreq, cfreq, cfreq).rfft()
+
+    # on the left: raw FFT magnitude
+    output['scalars'] = output.active_scalars.real
+    warped_raw = output.warp_by_scalar()
+
+    # on the right: scale to fixed warped height
+    output_scaled = output.translate((-11, 11, 0), inplace=False)
+    output_scaled['scalars_warp'] = output['scalars'] / abs(output['scalars']).ptp() * scalar_ptp
+    warped_scaled = output_scaled.warp_by_scalar('scalars_warp')
+    warped_scaled.active_scalars_name = 'scalars'
+    # push center back to xy plane due to peaks near 0 frequency
+    warped_scaled.translate((0, 0, -warped_scaled.center[-1]), inplace=True)
+
+    return warped_raw + warped_scaled
```

Comment:
I'm happy with this. It reads well (at least to me), and I don't think it can be refactored easily without losing readability.